test/results/protein_folding_tools/test_protein_xyz.py

```diff
@@ -0,0 +1,131 @@
+        """Tests the side positions list."""
+        for result, expected in zip(
+            self.test_xyz_1.side_positions, [None, None, None, None, None, None, None]
+        ):
+            if expected is None:
+                self.assertIsNone(result)
+            else:
+                self.assertTrue(np.allclose(result, expected, atol=1e-6))
+
+        for result, expected in zip(
+            self.test_xyz_2.side_positions,
+            [
+                None,
+                None,
+                np.array([0.57735027, 0.57735027, -1.73205081]),
+                np.array([2.30940108, -1.15470054, 0.0]),
+                None,
+            ],
+        ):
+            if expected is None:
+                self.assertIsNone(result)
+            else:
+                self.assertTrue(np.allclose(result, expected, atol=1e-6))
+
+    def test_main_positions(self):
+        """Tests the main position array."""
+        self.assertTrue(
+            np.allclose(
+                self.test_xyz_1.main_positions,
+                np.array(
+                    [
+                        [0.0, 0.0, 0.0],
+                        [0.57735027, 0.57735027, -0.57735027],
+                        [1.15470054, 0.0, -1.15470054],
+                        [1.73205081, -0.57735027, -0.57735027],
+                        [2.30940108, 0.0, 0.0],
+                        [1.73205081, 0.57735027, 0.57735027],
+                        [1.15470054, 1.15470054, 0.0],
+                    ]
+                ),
+                atol=1e-6,
+            )
+        )
+
+        self.assertTrue(
+            np.allclose(
+                self.test_xyz_2.main_positions,
+                np.array(
+                    [
+                        [0.0, 0.0, 0.0],
+                        [0.57735027, 0.57735027, -0.57735027],
+                        [1.15470054, 0.0, -1.15470054],
+                        [1.73205081, -0.57735027, -0.57735027],
+                        [2.30940108, 0.0, 0.0],
+                    ]
+                ),
+                atol=1e-6,
+            )
+        )
+
+    def test_xyz_file(self):
+        """Tests the array that will get converted in a xyz file."""
+        self.assertTrue(
+            (
+                self.test_xyz_1.get_xyz_file("", False)
+                == np.array(
+                    [
+                        ["A", "0.0", "0.0", "0.0"],
+                        ["P", "0.5773502691896258", "0.5773502691896258", "-0.5773502691896258"],
+                        ["R", "1.1547005383792517", "0.0", "-1.1547005383792517"],
+                        ["L", "1.7320508075688776", "-0.5773502691896258", "-0.5773502691896258"],
+                        ["R", "2.3094010767585034", "0.0", "0.0"],
+                        ["F", "1.7320508075688776", "0.5773502691896258", "0.5773502691896258"],
+                        ["Y", "1.154700538379252", "1.1547005383792517", "0.0"],
+                    ],
+                    dtype="<U32",
```

Comment:
I would extract the data into a variable called `expected`.
Please also check other places.